cmd/kcp-catalog/bind/catalogentry/cmd.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package catalogentry
+
+import (
+	"context"
+	"time"
+
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	catalogv1alpha1 "github.com/kcp-dev/catalog/api/v1alpha1"
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	pluginhelpers "github.com/kcp-dev/kcp/pkg/cliplugins/helpers"
+	"github.com/kcp-dev/logicalcluster/v2"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	"github.com/kcp-dev/kcp/pkg/cliplugins/base"
+	"k8s.io/client-go/rest"
+)
+
+// BindOptions contains the options for creating APIBindings for CE
+type BindOptions struct {
+	*base.Options
+	// WorkspacePath refers to the absolute reference to the Catalog Entry workspace
+	// to which the user wants to bind.
+	workspacePath string
+	// Name of the CE.
+	catalogentryName string
+	// APIExportRef is the argument accepted by the command. It contains the
```

Comment:
Need to decide if we're going to use WorkspacePath+CatalogEntryName, or CatalogEntryRef.

Reply:
For now the CatalogEntryRef is `<absolute_ref_to_workspace>:<catalog_entry>`. We fetch catalog entry from this workspace and then use ExportReference to find where apiexports are mapped to. Eventually the latter would be an encoded SHA string.